homeassistant/components/vicare/climate.py

```diff
@@ -205,7 +205,9 @@ def update(self) -> None:
                     "heating_curve_shift"
                 ] = self._circuit.getHeatingCurveShift()
 
-            self._attributes["vicare_modes"] = self._circuit.getModes()
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attributes["vicare_modes"] = self._circuit.getModes()
+            self._attributes["vicare_programs"] = self._circuit.getPrograms()
```

Comment:
This line is a new feature and should not be part of the bug-fix PR.
Please remove it so we can add this PR to  2024.3

Reply:
true

```suggestion
```